### ISSUE TITLE
Pass remote in sync() method

### ISSUE THREAD

**Issue by leplatrem** [Kinto/kinto.js#257]
When I built [this small demo](https://github.com/leplatrem/kinto-demo-leaflet), I felt puzzled because authorization `headers` can be passed to `sync()` but not the `remote` server. I would really have expected it there :) 

I looked at the code, and realized it was not an obvious change (because of api instantiation).

Would you find relevant if I spend some time on this?

**Comment by n1k0** [comment]
I hear you. I don't have a sense how critical is it though, but yeah, we could probably create the `Api` instance at `sync()` call time only. On the other hand, it's also cool being able to just call `sync()` without having to pass it the whole server configuration each time, but that's probably not that a big deal.

I'd love seeing a PR for that, but one more time, I'm not sure it's P1.

**Comment by leplatrem** [comment]
>  it's also cool being able to just call sync() without having to pass it

Oh yes of course! What I had in mind is to treat `remote` just like `headers`, i.e. being able to pass them in both places

**Comment by n1k0** [comment]
The more I'm working on the [amo-blocklist-ui](https://github.com/mozilla-services/amo-blocklist-ui), the more I realize we badly need to decouple anything server settings related from the `KintoBase` constructor, and only handle it from `#sync()`.

We should treat a Kinto instance as a local db driver, and collections an abstraction on top of tables. I'd go as far as extracting `#sync()` to its own separated module:

```js
import sync from "kinto/lib/sync";

const kinto = new Kinto();
const articles = kinto.collection("articles");

sync(articles, {
  remote: "http://localhost:9999",
  headers: {...}
});
```

We could also imagine having a `liveSync` version of it at some point:

```js
import liveSync from "kinto/lib/liveSync";

sync(liveSync, settings);
```

Having these things in their own module will greatly help avoiding bloating the `Collection` class, and will probably ease testing by a lot.

It will also help building a more powerful admin UI where server settings could be distincly provided for each collection.

Thoughts?

**Comment by leplatrem** [comment]
It reminds me an idea of @michielbdejong to be able to sync databases of existing applications that already store data in indexeddb.

Otherwise, as a quick feedback I am ok with this change. And in a first we are not obliged to break the current public API of the collection class.

> We could also imagine having a liveSync version of it at some point:

I didn't get that part, but I see the idea: depending of the nature of the first parameter, being able to handle different sync scenarios.

PouchDb has a slightly different approach but with the same idea. You instantiate a client on local or remote, and sync both instances:

```js
var localDB = new PouchDB('mylocaldb')
var remoteDB = new PouchDB('http://localhost:5984/myremotedb')
localDB.sync(remoteDB);
```

**Comment by n1k0** [comment]
> I didn't get that part

It was an example how we could enable live syncing through notifications, but we'll tackle this whole thing later.

**Comment by leplatrem** [comment]
Was this fixed via #276?

**Comment by n1k0** [comment]
Yup!
